content/applications/inventory_and_mrp/inventory/routes/concepts/stock_warehouses.rst

```diff
@@ -11,186 +11,214 @@ warehouses. In Odoo, pulling products from multiple warehouses to satisfy sales
-   under the :guilabel:`Other Info` tab on the sales order form *must* match in order for the
-   products included in the sales order to be pulled from different warehouses.
-
-  - If the virtual warehouse is not the value in the :guilabel:`Source Location` field on the
-    warehouse delivery form, then click :guilabel:`Edit`, make the change, and click
-    :guilabel:`Save`.
-  - If the virtual warehouse is not the value in the :guilabel:`Warehouse` field on the sales order,
-    then a new quotation may need to be generated.
-  - If whe :guilabel:`Warehouse` field is missing on the sales order form, then the virtual
-    warehouse (and its children warehouses) may not have been set up correctly, in which case,
-    review the documentation above again to make sure all settings/configuration were done properly.
-
-.. image:: stock_warehouses/stock-warehouses-delivery-order.png
+.. image:: stock_warehouses/delivery-order.png
    :align: center
    :alt: Delivery order with matching source and child locations.
 
-Finally, on the warehouse delivery form, under the :guilabel:`Detailed Operations` tab, confirm
-that the *Locations* values under the :guilabel:`From` column for each product matches to the *child
-locations* that are tied to the virtual *parent location*.
-
-.. note::
-   To view which *Locations* the products are coming from on the drop-down menus, click the
-   :guilabel:`internal link (arrow)` icon to expand the *Location* information. If needed, it can be
-   changed from here (granted there is quantity on hand for the product in that location).
-
-Once everything has been properly set, click :guilabel:`Validate` and then :guilabel:`Apply` to
-validate the delivery. Then, navigate back to the sales order form (via the breadcrumbs), and
-click :guilabel:`Create Invoice` to invoice for the sales order.
+.. important::
+   The :guilabel:`Source Location` on the warehouse delivery form and the :guilabel:`Warehouse`
+   under the :guilabel:`Other Info` tab on the sales order **must** match for products in the sales
+   order to be pulled from different warehouses.
+
+  - If the virtual warehouse is not in the :guilabel:`Source Location` field on the warehouse
+    delivery form, retry product reservation by:
+
+    - Running the scheduler: turn on :ref:`developer mode <developer-mode>`, and then go to
+      :menuselection:`Inventory app --> Operations --> Run Scheduler`
+    - Clicking :guilabel:`Check Availability` in the delivery order.
+  - If the virtual warehouse is **not** assigned to the :guilabel:`Warehouse` field on the sales
+    order, then cancel it and create a new sales order with the virtual warehouse set in the
+    :guilabel:`Warehouse` field.
+  - If the :guilabel:`Warehouse` field is missing on the sales order form, then the multiple
+    children warehouses may not have been set up correctly. Review the :ref:`previous section
+    <inventory/routes/child-wh>` to ensure the correct settings.
+
+.. danger::
```

Comment:
The danger section makes sense to me, but I think we could provide a bit more guidance to users who have two or three-step delivery enabled:
-What is the effect of having a virtual address set as the output/packing location?
-In what situations would a user still want to use this workflow?
-Is there any workaround that makes this flow work normally for two and three-step users?
-Etc.

Also, since this warning seems to be very important, should we put it at the top of the doc? I don't think we want a two or three-step user to get to the end of the workflow before realizing that they can't use this fulfillment method.